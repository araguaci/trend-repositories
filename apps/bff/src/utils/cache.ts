const LRU = require('lru-cache');

export let cache = null;

export const initCache = () => {
  const options = {
    max: 500,
    ttl: 86400 * 1000, // 1 day
  };

  cache = new LRU(options);
};
