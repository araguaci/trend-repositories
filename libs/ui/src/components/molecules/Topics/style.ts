import styled from 'styled-components';

export const StyledTopics = styled.div`
  display: block;
  align-items: center;
  margin-top: 7px;
  margin-bottom: 7px;

  & > a {
    margin-right: 7px;
  }
`;
