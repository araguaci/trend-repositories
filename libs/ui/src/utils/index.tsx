/* istanbul ignore file */
import { ThemeProvider } from 'styled-components';
import theme from '../theme';
import { ReactNode } from 'react';
import { render, act } from '@testing-library/react';
import { MockedProvider } from '@apollo/client/testing';
import { MockedResponse } from '@apollo/client/testing/core';
import { InMemoryCache } from '@apollo/client/cache';

export const renderWithProvider = (node: ReactNode) => {
  return render(<ThemeProvider theme={theme}>{node}</ThemeProvider>);
};

export const renderWithMockProvider = (
  node: ReactNode,
  cache: InMemoryCache,
  mocks: ReadonlyArray<MockedResponse>
) => {
  return render(
    <ThemeProvider theme={theme}>
      <MockedProvider cache={cache} mocks={mocks}>
        {node}
      </MockedProvider>
    </ThemeProvider>
  );
};

export const renderWithThemeProvider = (node: ReactNode) => {
  return render(<ThemeProvider theme={theme}>{node}</ThemeProvider>);
};

export const waitForResponse = async () =>
  await act(async () => await new Promise((resolve) => setTimeout(resolve, 0)));
export * from '@testing-library/react';
