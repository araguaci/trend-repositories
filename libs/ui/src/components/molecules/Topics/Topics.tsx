import { FC } from 'react';
import { Badge } from '@ui/atoms';
import { StyledTopics } from './style';
import { GithubRepository } from '@trend-repositories/api-interfaces';

type TopicsProps = Pick<GithubRepository, 'topics'>;

const Topics: FC<TopicsProps> = ({ topics }) => {
  return (
    <StyledTopics>
      {topics?.map((topic) => (
        <Badge
          key={topic}
          href={`https://github.com/search?q=topic:${topic}&type=Repositories`}
        >
          {topic}
        </Badge>
      ))}
    </StyledTopics>
  );
};

export default Topics;
