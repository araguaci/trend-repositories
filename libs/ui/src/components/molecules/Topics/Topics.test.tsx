import { renderWithThemeProvider, screen } from '@ui/utils';
import Topics from './Topics';

describe('Topics Component Test', () => {
  it('should be rendered correctly', () => {
    const { asFragment } = renderWithThemeProvider(
      <Topics topics={['topic1', 'topic2']} />
    );

    expect(
      screen.getByRole('link', { name: 'topic1' }).getAttribute('href')
    ).toEqual('https://github.com/search?q=topic:topic1&type=Repositories');

    expect(asFragment()).toMatchSnapshot();
  });
});
