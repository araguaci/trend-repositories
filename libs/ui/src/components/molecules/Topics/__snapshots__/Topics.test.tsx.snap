// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Topics Component Test should be rendered correctly 1`] = `
<DocumentFragment>
  <div
    class="style__StyledTopics-sc-19kmfo9-0 jwmWIf"
  >
    <a
      class="style__StyledLink-ikww7b-0 fUma-dV style__StyledBadge-o2pv9p-0 klpobS"
      href="https://github.com/search?q=topic:topic1&type=Repositories"
    >
      topic1
    </a>
    <a
      class="style__StyledLink-ikww7b-0 fUma-dV style__StyledBadge-o2pv9p-0 klpobS"
      href="https://github.com/search?q=topic:topic2&type=Repositories"
    >
      topic2
    </a>
  </div>
</DocumentFragment>
`;
